test/kernels/test_fidelity_statevector_kernel.py

```diff
@@ -343,6 +344,52 @@ def test_properties(self):
         self.assertEqual(qc, kernel.feature_map)
         self.assertEqual(1, kernel.num_features)
 
+    def test_pickling(self):
+        """Test that the kernel can be pickled correctly and without error."""
+        qc = QuantumCircuit(1)
+        qc.ry(Parameter("w"), 0)
+        kernel1 = FidelityStatevectorKernel(feature_map=qc)
+
+        pickled_obj = pickle.dumps(kernel1)
+        kernel2 = pickle.loads(pickled_obj)
+
+        kernel3 = FidelityStatevectorKernel()
+        kernel3.__setstate__(kernel1.__getstate__())
+
+        with self.subTest("Fail if objects 1 and 2 are not the same type"):
+            self.assertEqual(type(kernel1), type(kernel2))
+
+        with self.subTest("Fail if objects 1 and 3 are not the same type"):
+            self.assertEqual(type(kernel1), type(kernel3))
+
+        with self.subTest("Fail if objects 1 and 2 are not unique"):
+            self.assertNotEqual(kernel1, kernel2)
+
+        with self.subTest("Fail if objects 1 and 3 are not unique"):
+            self.assertNotEqual(kernel1, kernel3)
+
+        with self.subTest("Fail if caches 1 and 2 are incorrectly initialised"):
+            self.assertEqual(type(kernel1._get_statevector), type(kernel2._get_statevector))
+
+        with self.subTest("Fail if caches 1 and 3 are incorrectly initialised"):
+            self.assertEqual(type(kernel1._get_statevector), type(kernel3._get_statevector))
```

Comment:
To make testing clearer, can you please test messages more informative?

Reply:
When the tests fail, it would be hard to understand for the people, for example, where it is originated
Instead of `"Fail if objects 1 and 3 are not the same type"`, maybe, something like 
`"Pickle test failed, kernels are not the same type"`.

Reply:
Are these better 83a00d3 ? B)

Reply:
Sounds better, thanks